Sources/SwiftParser/ParseSourceFile.swift

```diff
@@ -152,8 +152,25 @@ extension Parser {
     }
 
     let remainingTokens = self.consumeRemainingTokens()
+
+    func parseEndOfFileIfPresent(raw: RawSyntax) -> R? {
+      if let endOfFileToken = self.consume(if: .endOfFile),
+        !endOfFileToken.leadingTriviaPieces.isEmpty,
+        let raw = raw.withTrailingTrivia(
+          Trivia(
+            rawPieces: endOfFileToken.leadingTriviaPieces
+          ),
+          arena: self.arena
+        )
+      {
+        return R.init(raw)!
+      } else {
+        return nil
```

Comment:
What about just returning `raw` here? That way you don’t need to do `??` on every call